software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,952 @@
+        return (self.__type == self.StatType.STAT)
+
+
+
+ 
+class VanillaStatsParser:
+    # Default coordinates of origin tile
+    _BSG_ORIGIN_X = 0
+    _BSG_ORIGIN_Y = 1
+
+    # formatting parameters for aligned printing
+    type_fmt = {"name"      : "{:<35}",
+                "type"      : "{:>20}",
+                "int"       : "{:>20}",
+                "float"     : "{:>20.4f}",
+                "percent"   : "{:>20.2f}",
+                "cord"      : "{:<2}, {:<31}",
+                "tag"       : "Tag {:<2}",
+               }
+
+
+    print_format = {"tg_timing_header": type_fmt["name"] + type_fmt["type"] + type_fmt["type"]    + type_fmt["type"]    + type_fmt["type"]    + type_fmt["type"]    + "\n",
+                    "tg_timing_data"  : type_fmt["name"] + type_fmt["int"]  + type_fmt["int"]     + type_fmt["float"]   + type_fmt["percent"] + type_fmt["percent"] + "\n",
+                    "timing_header"   : type_fmt["name"] + type_fmt["type"] + type_fmt["type"]    + type_fmt["type"]    + type_fmt["type"]    + type_fmt["type"]    + "\n",
+                    "timing_data"     : type_fmt["cord"] + type_fmt["int"]  + type_fmt["int"]     + type_fmt["float"]   + type_fmt["percent"] + type_fmt["percent"] + "\n",
+                    "instr_header"    : type_fmt["name"] + type_fmt["int"]  + type_fmt["type"]    + "\n",
+                    "instr_data"      : type_fmt["name"] + type_fmt["int"]  + type_fmt["percent"] + "\n",
+                    "stall_header"    : type_fmt["name"] + type_fmt["type"] + type_fmt["type"]    + type_fmt["type"]    + "\n",
+                    "stall_data"      : type_fmt["name"] + type_fmt["int"]  + type_fmt["percent"] + type_fmt["percent"] + "\n",
+                    "miss_header"     : type_fmt["name"] + type_fmt["type"] + type_fmt["type"]    + type_fmt["type"]    + "\n",
+                    "miss_data"       : type_fmt["name"] + type_fmt["int"]  + type_fmt["int"]     + type_fmt["float"]   + "\n",
+                    "tag_header"      : type_fmt["name"] + type_fmt["type"] + type_fmt["type"]    + type_fmt["type"]    + type_fmt["type"]    + type_fmt["type"] + "\n",
+                    "tag_data"        : type_fmt["name"] + type_fmt["int"]  + type_fmt["int"]     + type_fmt["int"]     + type_fmt["float"]   + type_fmt["percent"] + "\n",
+                    "tag_separator"   : '-' * 75 + ' ' * 2 + type_fmt["tag"]  + ' ' * 2 + '-' * 75 + "\n",
+                    "start_lbreak"    : '=' *160 + "\n",
+                    "end_lbreak"      : '=' *160 + "\n\n",
+                   }
+
+
+
+    # default constructor
+    def __init__(self, manycore_dim_y, manycore_dim_x, per_tile_stat, per_tile_group_stat, input_file):
+
+        self.manycore_dim_y = manycore_dim_y
+        self.manycore_dim_x = manycore_dim_x
+        self.manycore_dim = manycore_dim_y * manycore_dim_x
+        self.per_tile_stat = per_tile_stat
+        self.per_tile_group_stat = per_tile_group_stat
+
+        self.max_tile_groups = 1 << 14
```

Comment:
You should reuse the _TG_ID_WIDTH class variable from Cuda Stat Tag.

Reply:
good idea, thanks, done!